tasks/update-cr-status/update-cr-status.yaml

```diff
@@ -0,0 +1,52 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: update-cr-status
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    A tekton task that updates the passed CR status with the
+    contents stored in the resultsFile.
+  params:
+    - name: resourceType
+      description: The type of resource that is being patched
+      type: string
+      default: release
+    - name: resource
+      description: The namespaced name of the resource to be patched
+      type: string
+    - name: resultsFilePath
+      description: |
+        Path to the JSON string of the results in the data workspace which will be added to the resource's status
+      type: string
+  workspaces:
+    - name: data
+      description: Workspace where the results file is stored
+  steps:
+    - name: update-cr-status
+      image: quay.io/redhat-appstudio/release-service-utils:bc81bfed6062a386e48a76b252c6f33b52c411b0
+      script: |
+        #!/usr/bin/env bash
+        set -ex
+
+        RESULTS_FILE="$(workspaces.data.path)/$(params.resultsFilePath)"
+        if [ ! -f "${RESULTS_FILE}" ] ; then
+            echo "No results JSON file was found."
+            exit 0
```

Comment:
Why wouldn't this fail? What's the use case for this?

Reply:
I wrote it this way in case we add it to every pipeline we have. It is feasible some of them do not generate results file. Or, if we put this task in the finally and the pipeline fails before a results file is generated - we would not want the finally block to fail on this

Reply:
Multiple tasks will generate the file. You can check the epic for the ones that will for now, but like push-snapshot will, create-advisory will. Let's say we fail after push-snapshot but on something else. I would still expect the results from push-snapshot end up in the release status. So I think it makes sense in the finally. However, I still want the cleanup-workspace task to run always, so I don't this task to fail the finally block. Hopefully I am making sense

Reply:
I guess I forgot about the fact that this is mostly to save some urls, but the success status - that part is done in the operator. So yeah, it makes sense.